sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -798,9 +798,19 @@ multipartIdentifierList
     ;
 
 multipartIdentifier
-    : parts+=errorCapturingIdentifier (DOT parts+=errorCapturingIdentifier)*
+    : parts+=errorCapturingIdentifierWithTemplate (DOT parts+=errorCapturingIdentifierWithTemplate)*
     ;
 
+errorCapturingIdentifierWithTemplate
+   : identifierClause
+   | ident = errorCapturingIdentifier
+   ;
+
+errorCapturingSingleIdentifierWithTemplate
```

Comment:
what's the difference between this and `errorCapturingIdentifierWithTemplate`?

Reply:
The SingleIdentifier versions raise an error if the string literal passed to IDENTIFIER() parses as a qualified name.
They also produce a String. Where the others produce a Seq[String]